nixos/modules/services/web-apps/alps.nix

```diff
@@ -0,0 +1,99 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.alps;
+in {
+  options.services.alps = {
+    enable = mkEnableOption "alps";
+
+    port = mkOption {
+      type = types.port;
+      default = 1323;
+      description = ''
+        TCP port the process should listen to.
+      '';
+    };
+
+    bindIP = mkOption {
+      default = "[::]";
+      type = types.str;
+      description = ''
+        IP the service should listen to.
+      '';
+    };
+
+    theme = mkOption {
+      type = types.enum [ "alps" "sourcehut" ];
+      default = "sourcehut";
+      description = ''
+        The frontend's theme.
+      '';
+    };
+
+    imaps = {
+      port = mkOption {
+        type = types.port;
+        default = 993;
+        description = ''
+          The imaps port of the server
+        '';
+      };
+
+      host = mkOption {
+        type = types.str;
+        default = "[::1]";
+        example = "mail.example.org";
+        description = ''
+          The imaps server's address.
```

Comment:
I think that `imaps` was actually reasonable to state that this only works with that.

Reply:
The config he produces only supports IMAP with SSL so the imaps makes sense because its the secure version.

My suggestion would be: `The address of the secure IMAP server`.

Reply:
I opted to use the formulation with IMAPs, I hope this is agreeable.